### PR TITLE
Bringing build/test/lint setup up to our modern standards. Will fix l…

### PR DESCRIPTION
**Pull request by kenwheeler**
…int in a later PR for clarity.

- Removed gulp
- Using elint-config-defaults
- Got tests running
- Fixed some prop based React warnings

### CODE REVIEW

**Comment by kenwheeler** [comment]
cc @baer @alexlande

**Comment by alexlande** [comment]
Rad :+1:

File: .gitignore [hunk 1]
```diff
@@ -1,54 +1,35 @@
...
 Icon
+
+# Thumbnails
 ._*
+
+# Files that might appear on external disk
 .Spotlight-V100
 .Trashes
 
-# --------------------
```
Comment (baer) [inline .gitignore]: Why are you removing so much of the gitignore? This is the same file we use elsewhere and it is a good baseline for multiple environments both windows and mac.

Comment (kenwheeler) [inline .gitignore]: I wanted it to match our component boilerplate .gitignore for consistency, but do you think I should instead add these lines to our boilerplate?

Comment (baer) [inline .gitignore]: Meh, its up to you but I'd vote yes. IMO its in our best interest to make contribution as easy as possible no matter what env/IDE you use. If you want to tune for our use case that is fair too. Just my two cents.

Just to give you all the background, this is a very small subset that was pulled from Github's gitignore repo
https://github.com/github/gitignore

**Pull request merged**